src/Compiler/Scheme/Gambit.idr

```diff
@@ -401,10 +400,8 @@ compileExpr c s tmpDir outputDir tm outfile
                  Nothing => gscBackend ++ ["-exe", "-cc-options", "-Wno-implicit-function-declaration", "-ld-options"] ++ libsfile
                  Just _ => ["-c"]
          let cmd = gsc ++ gscCompileOpts ++ ["-o", execPath, srcPath]
-         ok <- coreLift $ system cmd
-         if ok == 0
-            then pure (Just execPath)
-            else pure Nothing
+         safeSystem cmd
+         pure (Just execPath)
```

Comment:
This is not logically equivalent. Why is it legitimate to drop the `Nothing` branch?

Reply:
`safeSystem` throws an error if the exit code is not zero. This makes the errors more clear